impit-node/src/response.rs

```diff
@@ -1,23 +1,39 @@
-use std::collections::HashMap;
+use std::{cell::RefCell, collections::HashMap, ops::Deref};
 
 use impit::utils::{decode, ContentType};
-use napi::{bindgen_prelude::Buffer, Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi::{bindgen_prelude::{Buffer, ReadableStream, Result, BufferSlice}, Env, JsFunction, JsObject, JsString, JsUnknown};
 use napi_derive::napi;
+use tokio_stream::{wrappers::ReceiverStream, StreamExt};
 use reqwest::Response;
 
 #[napi]
 pub struct ImpitResponse {
-  bytes: Vec<u8>,
+  bytes: RefCell<Option<Vec<u8>>>,
+  inner: RefCell<Option<Response>>,
```

Comment:
Don't you always assign this in `from`? It kinda looks like the whole `RefCell<Option<..>>` is redundant. Or is it there so that you can borrow it mutably in `stream_body`?

Reply:
Yes, the `RefCell` is for the mutable borrows in functions without mutable `self`. `napi` enforces stricter rules on mutability and borrows because of (I'm assuming) sharing the data between different environments. 

E.g. asynchronous functions with `mut& self` are banned altogether (won't compile), same with mutating functions with Env argument ([docs](https://napi.rs/docs/concepts/inject-env)), for usage see e.g. the `json` function on response.

bytes are `Option<>` bcs sometimes, we haven't read the response yet. `inner` is `Option<>` so we can consume it with the `bytes` or `bytes_stream` call and we don't end up with a `RefCell` with undefined value.


There could absolutely be a better solution to all this. If you see one, I'm all ears!

Reply:
The response body is now stored in a JS `Response` instance, this solution has been dropped completely. Resolving this convo for now.